src/agent.rs

```diff
@@ -112,16 +126,17 @@ impl Agent {
                 })
             }
             STATUS => {
-                let err = rxf.try_recv().unwrap();
-                info!("Failed {}", err);
-                Err(err)
+                let output = rxf.try_recv().unwrap();
+                info!("Failed {}", output.status);
+                //TODO: Potential Error output is lost here. Should be fixed.
```

Comment:
Do you want to fix this? If not, please file an issue and link to it here.

Reply:
fixed.